Release 2.1.0

csyhuang opened the pull request:
Hi @chpolste ,

I am submitting this PR to do a formal release v2.1.0, as all the items in this release ticket  #117 has been done. I have tested the additional components and rerun the jupyter notebooks to ensure everything works. Please review only if you have time (esp. on the changes in `xarrayinterface`), as this is not a huge change from v2.1.0a. 

I will make the release by March 17 (since that's the deadline for resubmitting our paper, and this release is mentioned in the updated text). Thanks a lot!

Cheers,

Clare

chpolste reviewed (approved):
Hi @csyhuang, sorry for the late review. This looks good to me :)

I couldn't test the new notebooks without the MERRA2 data. Some docs suggestions in #136 and #135 if you want to tick off the dimension-check TODO already now (please test the updated loops with data though first if you have it ready - or leave the PR until v2.2).

Remember to also remove the `a` of the version in the `pyproject.toml`. Always annoying when the version string has to live in so many places.

csyhuang commented:
Hi Christopher @chpolste Great thanks for reviewing the PR and all the edits! 😄

I added two more files into this PR for easier iteration for the next release:

- `docs/source/Memo for Developers.rst`: a memo for our use to note down things that are easily forgotten (e.g. location of version number)
- `tests/test_integration.py`: I added sampled MERRA2 data and an integration for easier debugging locally - hopefully this helps with the iteration of #135 

I will merge this PR and make the release later today. Let's continue the conversation regarding #135 for release 2.2 #133 .

Cheers,

Clare
